### PR TITLE
Fixes for issue #289, #290, #291

### PR DESCRIPTION
**Pull request by mukul-ms**
#### Following fixes were made
- Links are rendered properly inside `Rule Description`
     image
- Spacing + newline issues between paragraphs in `Rule Description`
    image
- Instead of showing both short and full description, we do:
    - If full description is present, show full description
    - If full description is not present but short description is, show short description
    - If short description is not present, show empty state

### CODE REVIEW

File: src/panel/details.tsx [hunk 1]
```diff
@@ -109,34 +109,34 @@ interface DetailsProps { result: Result, height: IObservableValue<number> }
...
                                     <span className="svSecondary">No stacks in selected result.</span>
-                                </div>
+                                </div>;
 
                             return this.stacks.map(stack => {
-                                const stackFrames = stack.frames
+                                const stackFrames = stack.frames;
 
-                                const selection = observable.box(undefined as Location, { deep: false })
+                                const selection = observable.box(undefined as unknown as Location, { deep: false });
```
Comment (jeffersonking) [inline src/panel/details.tsx:135]: Likely `observable.box<Location | undefined>(undefined.....`

File: src/panel/details.tsx [hunk 2]
```diff
@@ -36,47 +36,47 @@ interface DetailsProps { result: Result, height: IObservableValue<number> }
...
                                 ? <ReactMarkdown className="svMarkDown" source={result._markdown} escapeHtml={false} />
-                                : renderMessageWithEmbeddedLinks(result, vscode.postMessage)}</div>
+                                : renderMessageWithEmbeddedLinks(result, vscode.postMessage, result._message)}</div>
                         <div className="svDetailsGrid">
                             <span>Rule Id</span>			{helpUri ? <a href={helpUri} target="_blank" rel="noopener noreferrer">{result.ruleId}</a> : <span>{result.ruleId}</span>}
                             <span>Rule Name</span>			<span>{result._rule?.name ?? '—'}</span>
-                            <span>Rule Desc Short</span>	<span>{renderRuleDesc(result._rule?.shortDescription)}</span>
-                            <span>Rule Desc Full</span>		<span>{renderRuleDesc(result._rule?.fullDescription)}</span>
+                            <span>Rule Desc Short</span>	<span><div>{renderRuleDesc(result, result._rule?.shortDescription)}</div></span>
+                            <span>Rule Desc Full</span>		<span><div>{renderRuleDesc(result, result._rule?.fullDescription)}</div></span>
```
Comment (jeffersonking) [inline src/panel/details.tsx:79]: Reason for new `div`?

File: src/panel/details.tsx [hunk 3]
```diff
@@ -36,47 +36,47 @@ interface DetailsProps { result: Result, height: IObservableValue<number> }
         });
     }
     render() {
-        const renderRuleDesc = (desc?: { text: string, markdown?: string }) => {
+        const renderRuleDesc = (result: Result, desc?: { text: string, markdown?: string }) => {
             if (!desc) return '—';
             return desc.markdown
-                ? <ReactMarkdown className="svMarkDown" source={desc.markdown} />
-                : desc.text;
+                ? <ReactMarkdown className="svMarkDown" source={desc.markdown} escapeHtml={false} />
```
Comment (jeffersonking) [inline src/panel/details.tsx:42]: What trigged the addition of `escapeHtml`?

**Comment by jeffersonking** [comment]
Concerned about the readability/usability/entanglement of `renderRuleDesc` and `renderMessageWithEmbeddedLinks`. Thinking about if there's a better way.

File: src/panel/details.tsx (hunk 2 above)
Comment (mukul-ms) [reply src/panel/details.tsx]: Was having this issue: https://stackoverflow.com/questions/52529682/how-do-you-make-links-href-in-css-grid-appear-inline-versus-block

File: src/panel/details.tsx (hunk 3 above)
Comment (mukul-ms) [reply src/panel/details.tsx:43]: It was for `svDetailsMessage` but not for `renderRuleDesc`

Comment (jeffersonking) [reply src/panel/details.tsx:43]: Great!

File: src/panel/details.tsx (hunk 2 above)
Comment (mukul-ms) [reply src/panel/details.tsx]: Thanks. Fixed

File: src/panel/details.tsx (hunk 1 above)
Comment (mukul-ms) [reply src/panel/details.tsx]: Fixed

**Pull request merged**